src/docker.rs

```diff
@@ -187,4 +199,31 @@ WORKDIR /home/solana
         )?;
         Ok(())
     }
+
+    pub fn push_image(docker_image: &DockerImage) -> Result<(), Box<dyn Error>> {
+        let progress_bar = new_spinner_progress_bar();
+        progress_bar.set_message(format!(
+            "{ROCKET}Pushing {} image to registry...",
+            docker_image.validator_type()
+        ));
+        let command = format!("docker push '{}'", docker_image);
+        let output = match Command::new("sh")
+            .arg("-c")
+            .arg(&command)
+            .stdout(Stdio::null())
+            .stderr(Stdio::null())
+            .spawn()
+            .expect("Failed to execute command")
+            .wait_with_output()
+        {
+            Ok(res) => Ok(res),
+            Err(err) => Err(Box::new(err) as Box<dyn Error>),
+        }?;
```

Comment:
nit: you should be able to simplify this to something like:

```
        let output = match Command::new("sh")
            .arg("-c")
            .arg(&command)
            .stdout(Stdio::null())
            .stderr(Stdio::null())
            .spawn()
            .expect("Failed to execute command")
            .wait_with_output()
            .map_err(Box::new)?;
```

with maybe some trickery on the `Box::new` bit